sync/src/relayer/compact_block_process.rs

```diff
@@ -90,37 +94,31 @@ impl<'a, CS: ChainStore> CompactBlockProcess<'a, CS> {
                     },
                     Arc::clone(&self.relayer.shared.consensus().pow_engine()),
                 );
-                match header_verifier.verify(&resolver) {
-                    Ok(_) => {
-                        let ret = {
-                            let chain_state = self.relayer.shared.chain_state().lock();
-                            self.relayer.request_proposal_txs(
-                                &chain_state,
-                                self.nc,
-                                self.peer,
-                                &compact_block,
-                            );
-                            self.relayer
-                                .reconstruct_block(&chain_state, &compact_block, Vec::new())
-                        };
-                        match ret {
-                            Ok(block) => {
-                                self.relayer
-                                    .accept_block(self.nc, self.peer, &Arc::new(block))
-                            }
-                            Err(missing) => {
-                                missing_indexes = missing;
-                                pending_compact_blocks
-                                    .insert(block_hash.clone(), compact_block.clone());
-                            }
-                        }
-                    }
-                    Err(err) => {
-                        debug!(target: "relay", "unexpected header verify failed: {}", err);
-                    }
+                let compact_block_verifier = CompactBlockVerifier::new();
+                header_verifier.verify(&resolver)?;
```

Comment:
Return error here will ban this peer, but verify failed happens frequently (UnknownParent error for example)

Reply:
It's my fault.

Errors for invalid block format should be banned, 
some errors like `UnknownParent` should be warned.

I will change here to log errors without returning outside. And refactoring score system later is required, so that we can handle errors gracefully.

Reply:
Yes, we need to dispatch different errors to different punishment. Maybe add a trait Punish on the errors.

Reply:
Updated.

But error thrown from `CompactBlockVerifier` is still return outside. The reason is that this kind of errors must be invalid format error, we can ban it.